jupiterone/resource_framework.go

```diff
@@ -0,0 +1,297 @@
+		if err != nil {
+			diag.AddError(fmt.Sprintf("Could not marshal scope filter at index %d", i), err.Error())
+		}
+	}
+	return scopeFilters, diag
+}
+
+var _ resource.Resource = &ComplianceFrameworkResource{}
+var _ resource.ResourceWithConfigure = &ComplianceFrameworkResource{}
+var _ resource.ResourceWithImportState = &ComplianceFrameworkResource{}
+var _ resource.ResourceWithConfigValidators = &ComplianceFrameworkResource{}
+var _ resource.ResourceWithModifyPlan = &ComplianceFrameworkResource{}
+
+type ComplianceFrameworkResource struct {
+	version string
+	qlient  graphql.Client
+}
+
+func NewFrameworkResource() resource.Resource {
+	return &ComplianceFrameworkResource{}
+}
+
+// Metadata implements resource.Resource
+func (*ComplianceFrameworkResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_framework"
+}
+
+// Configure implements resource.ResourceWithConfigure
+func (r *ComplianceFrameworkResource) Configure(_ context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	// Prevent panic if the provider has not been configured.
+	if req.ProviderData == nil {
+		return
+	}
+
+	p, ok := req.ProviderData.(*JupiterOneProvider)
+
+	if !ok {
+		resp.Diagnostics.AddError(
+			"Unexpected Resource Configure Type",
+			fmt.Sprintf("Expected JupiterOneProvider, got: %T. Please report this issue to the provider developers.", req.ProviderData),
+		)
+
+		return
+	}
+
+	r.version = p.version
+	r.qlient = p.Qlient
+}
+
+// Schema implements resource.Resource
+func (*ComplianceFrameworkResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "A custom compliance standard, benchmark, or questionnaire.",
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed: true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"name": schema.StringAttribute{
+				Required:    true,
+				Description: "The framework's name",
+			},
+			"version": schema.StringAttribute{
+				Required:    true,
+				Description: "Version of the framework itself (not an J1 API version)",
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"framework_type": schema.StringAttribute{
+				Required:    true,
+				Description: "Whether this is a standard, benchmark, or questionnaire",
+				Validators: []validator.String{
+					stringvalidator.OneOf(FrameworkTypes...),
+				},
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"web_link": schema.StringAttribute{
+				Optional:    true,
+				Description: "A URL for referencing additional information about the framework",
+				// TODO: basic URL pattern matching
+				//Validators: []validator.String{
+				//},
```

Comment:
I'll make an issue for this.

Reply:
:+1: to file an issue. The API will accept a lot more than the web UI, so it's unclear to me how much the terraform provider itself should be enforcing.